pkg/secrets/vault_test.go

```diff
@@ -35,3 +40,127 @@ func TestValidateVaultConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestNoSecret(t *testing.T) {
+	notASecret := "notASecret"
+	eng, err := NewDecrypter(context.TODO(), notASecret)
+	assert.Nil(t, err)
+
+	unchanged, _ := eng.Decrypt()
+	assert.Equal(t, notASecret, unchanged)
+}
+
+func TestNoVaultConfig(t *testing.T) {
+	e := Engines["vault"]
+	delete(Engines, "vault")
+	decrypter, err := NewDecrypter(context.TODO(), "encrypted:vault!e:secret!n:test-secret!k:foo")
+	assert.NotNil(t, err)
+	assert.Nil(t, decrypter)
+	Engines["vault"] = e
+}
+
+func TestClientGetsNamespace(t *testing.T) {
+	cases := map[string]struct {
+		cfg VaultConfig
+	}{
+		"with namespace": {
+			cfg: VaultConfig{
+				Enabled:    true,
+				Url:        "http://127.0.0.1:8200",
+				AuthMethod: "KUBERNETES",
+				Namespace:  "my-namespace",
+				Role:       "my-role",
+				Path:       "my-path",
+				Token:      "my-token",
+			},
+		},
+		"no namespace": {
+			cfg: VaultConfig{
+				Enabled:    true,
+				Url:        "http://127.0.0.1:8200",
+				AuthMethod: "KUBERNETES",
+				Namespace:  "",
+				Role:       "my-role",
+				Path:       "my-path",
+				Token:      "my-other-token",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		d := VaultDecrypter{
+			vaultConfig: c.cfg,
+		}
+
+		client, err := d.FetchVaultClient(c.cfg.Token)
+		assert.Nil(t, err)
+
+		header := client.Headers()
+		if header != nil {
+			assert.Equal(t, c.cfg.Namespace, header.Get(consts.NamespaceHeaderName))
+		}
+
+		assert.Equal(t, c.cfg.Token, client.Token())
+	}
+}
+
+func TestParseVaultSecret(t *testing.T) {
+	cases := map[string]struct {
+		params      string
+		expected    func(*testing.T, *VaultDecrypter)
+		shouldError bool
+	}{
+		"standard syntax": {
+			"e:engine!p:path!k:key",
+			func(t *testing.T, decrypter *VaultDecrypter) {
+				assert.Equal(t, "engine", decrypter.engine)
+				assert.Equal(t, "path", decrypter.path)
+				assert.Equal(t, "key", decrypter.key)
+			},
+			false,
+		},
+		"with deprecated n param": {
+			"e:engine!n:path!k:key",
+			func(t *testing.T, decrypter *VaultDecrypter) {
+				assert.Equal(t, "engine", decrypter.engine)
+				assert.Equal(t, "path", decrypter.path)
+				assert.Equal(t, "key", decrypter.key)
+			},
+			false,
+		},
+		"with binary file flag": {
+			"e:engine!p:path!k:key!b:true",
+			func(t *testing.T, decrypter *VaultDecrypter) {
+				assert.Equal(t, "engine", decrypter.engine)
+				assert.Equal(t, "path", decrypter.path)
+				assert.Equal(t, "key", decrypter.key)
+				assert.Equal(t, "true", decrypter.base64Encoded)
+			},
+			false,
+		},
+		"missing engine ": {
+			"n:path!k:key",
+			func(t *testing.T, decrypter *VaultDecrypter) {},
+			true,
+		},
+		"missing path": {
+			"e:engine!k:key",
+			func(t *testing.T, decrypter *VaultDecrypter) {},
+			true,
+		},
+		"missing key": {
+			"e:engine!p:path",
+			func(t *testing.T, decrypter *VaultDecrypter) {},
+			true,
+		},
+	}
+
+	for _, c := range cases {
+		v := &VaultDecrypter{}
```

Comment:
Since you have a test name now, you can to `for testName, c := range cases` and then use it by wrapping the test in `t.Run(testName, func(t *testing.T) {}`.